src/runtime/hashmap.go

```diff
@@ -12,12 +12,30 @@ import (
 
 // The underlying hashmap structure for Go.
 type hashmap struct {
-	next       *hashmap       // hashmap after evacuate (for iterators)
 	buckets    unsafe.Pointer // pointer to array of buckets
 	count      uintptr
 	keySize    uint8 // maybe this can store the key type as well? E.g. keysize == 5 means string?
 	valueSize  uint8
 	bucketBits uint8
+	alg        uint8
+}
+
+type hashmapAlgorithm uint8
+
+const (
+	hashmapAlgorithmBinary hashmapAlgorithm = iota + 1
```

Comment:
Why not start at 0? It would avoid the zero entry in `hashmapAlgorithms` (and possibly make the rest of the code very slightly smaller).

Reply:
Because I initially wanted to catch places where `alg` wasn't initialized and having the nil function pointers would make it obvious.

I'll remove it since I'm more confident I'm catching it now.  (I used to initialize the fields in the new map by hand, but now I'm just copying the entire struct and replacing the fields I care about.)